### PR TITLE
pie-microsite@v1.35.0 Typography Page - List and group alias tokens on alias page

### PR DESCRIPTION
**Pull request by raoufswe**
### Added
- Typography alias tokens on the alias page, and group them as per the Figma designs
- A normalization layer to pie design tokens to group font alias tokens into wide/narrow
- DisplayName logic to the token metadata

### CODE REVIEW

File: apps/pie-microsite/src/_data/normalizeTokens.js
Comment (jamieomaguire) [inline apps/pie-microsite/src/_data/normalizeTokens.js:29]: question here - if the module.exports was just to return `normalizeFontAliasTokens()` (provided the function returned the new object), would that only be called once during the build or multiple times?

If it's only called once, it might be cleaner to go with that approach rather than calling the function in the module outside of any function but this is just personal preference.

Comment (jamieomaguire) [reply apps/pie-microsite/src/_data/normalizeTokens.js:29]: yes perfect - that's what I was thinking and poorly explained! :)

File: apps/pie-microsite/src/assets/styles/components/_tokensTable.scss
Comment (ashleynolan) [inline apps/pie-microsite/src/assets/styles/components/_tokensTable.scss:127]: Think this PR needs a rebase against the main branch – looks like it's got some changes that Lizzie's PR had in it?

Comment (raoufswe) [reply apps/pie-microsite/src/assets/styles/components/_tokensTable.scss]: yep it needed, i just rebased 💯

File: apps/pie-microsite/src/assets/styles/components/_contentPage.scss
Comment (ashleynolan) [inline apps/pie-microsite/src/assets/styles/components/_contentPage.scss:83]: It looks like the default `h2` size is wrong in fozzie (which is why you need to add these styles, as the microsite pulls in fozzie). We likely need an update ticket to change this in fozzie so it's up-to-date with the [latest foundations](https://www.figma.com/file/8LrMTRRHfQYSRZrK4cLoKs/%E2%9A%A1%EF%B8%8F%F0%9F%A7%8A-PIE-2.0-Foundations-%E2%80%93-Light-%F0%9F%A7%8A%E2%9A%A1%EF%B8%8F?node-id=5437%3A20693&t=o9F7a6rkwdFwU7jC-4).

Shouldn't hold up this ticket, but good that this has brought inconsistency up 👍

**Review by ashleynolan — APPROVED** [review]

File: apps/pie-microsite/src/assets/styles/components/_contentPage.scss
Comment (raoufswe) [reply apps/pie-microsite/src/assets/styles/components/_contentPage.scss:83]: I think @jamieomaguire has created a ticket for this update as well. we will relook at it in our next refinement hopefully. 💯

**Review by FayeCarter — DISMISSED** [review]

**Review by LTurns — DISMISSED** [review]

File: apps/pie-microsite/CHANGELOG.md
Comment (xander-marjoram) [inline apps/pie-microsite/CHANGELOG.md:14]: Should this be incremented here, and in package.json?

File: apps/pie-microsite/CHANGELOG.md
Comment (xander-marjoram) [inline apps/pie-microsite/CHANGELOG.md:16]: And while you're here, this date can be bumped

File: apps/pie-microsite/src/_11ty/shortcodes/tokensTable/index.js
Comment (xander-marjoram) [inline apps/pie-microsite/src/_11ty/shortcodes/tokensTable/index.js:115]: Looks like this is missing the example

File: apps/pie-microsite/src/_11ty/shortcodes/tokensTable/index.js
Comment (xander-marjoram) [inline apps/pie-microsite/src/_11ty/shortcodes/tokensTable/index.js:123]: A bit of indentation here might help readability

File: apps/pie-microsite/src/_11ty/shortcodes/tokensTable/index.js
Comment (xander-marjoram) [inline apps/pie-microsite/src/_11ty/shortcodes/tokensTable/index.js:127]: What does this filter do? Looks like it might filter out the text decoration if it isn't defined?

File: apps/pie-microsite/src/_11ty/shortcodes/tokensTable/index.js
Comment (xander-marjoram) [inline apps/pie-microsite/src/_11ty/shortcodes/tokensTable/index.js:129]: Sneaky semicolon
```suggestion
    return `<div class="c-tokensTable-example--font" style="${cssVariables.join('; ')}">String</div>`;
```

File: apps/pie-microsite/src/_11ty/shortcodes/tokensTable/index.js
Comment (xander-marjoram) [inline apps/pie-microsite/src/_11ty/shortcodes/tokensTable/index.js:141]: Nit: alphabetical?

File: apps/pie-microsite/CHANGELOG.md
Comment (raoufswe) [reply apps/pie-microsite/CHANGELOG.md:14]: yep will be done before merging 💯

File: apps/pie-microsite/src/_11ty/shortcodes/tokensTable/index.js
Comment (raoufswe) [reply apps/pie-microsite/src/_11ty/shortcodes/tokensTable/index.js:127]: yep, the filter is needed to filter undefined/falsy values from the array as not all examples/tokens need/have `text-decoration styles. 

`passing filter(Boolean)` is a neat trick to do so. 

https://michaeluloth.com/filter-boolean

File: apps/pie-microsite/src/_11ty/shortcodes/tokensTable/index.js
Comment (raoufswe) [reply apps/pie-microsite/src/_11ty/shortcodes/tokensTable/index.js:141]: 💯

File: apps/pie-microsite/src/_data/normalizeTokens.js
Comment (xander-marjoram) [inline apps/pie-microsite/src/_data/normalizeTokens.js:3]: Is this necessary if the original import comes directly from a file and is only used once?

File: apps/pie-microsite/src/_data/normalizeTokens.js
Comment (xander-marjoram) [inline apps/pie-microsite/src/_data/normalizeTokens.js:27]: This comment doesn't seem necessary yet

File: apps/pie-microsite/src/_data/tokenTypes.js
Comment (xander-marjoram) [inline apps/pie-microsite/src/_data/tokenTypes.js:10]: Nit: alphabetical please

File: apps/pie-microsite/src/content/pages/foundations/typography/tokens/alias/toggled-content/wide-screen-alias.content.md
Comment (xander-marjoram) [inline apps/pie-microsite/src/content/pages/foundations/typography/tokens/alias/toggled-content/wide-screen-alias.content.md:4]: The indentation looks a little inconsistent on these pages

File: apps/pie-microsite/src/_data/normalizeTokens.js
Comment (raoufswe) [reply apps/pie-microsite/src/_data/normalizeTokens.js:3]: are you referring to the cloning bit? if so, I think yes, even if the file is used once (so far) and imported from a file. the object will still have the same reference in memory of the original import and any mutation that happens will be carried to the original tokens.json. 

here is a [sandbox](https://codesandbox.io/s/laughing-hamilton-4dtmkt?file=/src/index.js) with some of the issues we might face down the road where the normalization layer leaks its mutation to other modules if it wasn't cloned.

File: apps/pie-microsite/src/_11ty/shortcodes/tokensTable/index.js
Comment (xander-marjoram) [reply apps/pie-microsite/src/_11ty/shortcodes/tokensTable/index.js:127]: I think we would typically do something like this to avoid the need to filter, but your way is probably nicer here:
```
const cssVariables = [
    `--example-font-size: ${fontSize}px`,
    `--example-font-line-height: ${lineHeight}px`,
    `--example-font-weight: ${fontWeight}`,
    ...(textDecoration ? [`--example-font-text-decoration: ${textDecoration}`] : [])
];
```

File: apps/pie-microsite/src/_data/normalizeTokens.js
Comment (xander-marjoram) [reply apps/pie-microsite/src/_data/normalizeTokens.js:3]: Well that's not ideal!

File: apps/pie-microsite/src/tokensMetadata.json
Comment (xander-marjoram) [inline apps/pie-microsite/src/tokensMetadata.json:831]: Nit
```suggestion
          "description": "Used for captions, labels, disclaimers or helper text. Use sparingly."
```

File: apps/pie-microsite/src/tokensMetadata.json
Comment (xander-marjoram) [inline apps/pie-microsite/src/tokensMetadata.json:714]: Nit:
```suggestion
          "description": "Used for captions, labels, disclaimers or helper text. Use sparingly."
```

**Review by jamieomaguire — DISMISSED** [review]

**Review by FayeCarter — DISMISSED** [review]

File: apps/pie-microsite/src/_11ty/shortcodes/tokensTable/index.js
Comment (LTurns) [inline apps/pie-microsite/src/_11ty/shortcodes/tokensTable/index.js:44]: super minor - wondering if it's worth adding a text-decoration token value in @param too since it's needed in this function 👍

File: apps/pie-microsite/src/_11ty/shortcodes/tokensTable/index.js
Comment (LTurns) [inline apps/pie-microsite/src/_11ty/shortcodes/tokensTable/index.js:141]: nit: indentation 😄

**Review by LTurns — APPROVED** [review]

File: apps/pie-microsite/src/_11ty/shortcodes/tokensTable/index.js
Comment (xander-marjoram) [reply apps/pie-microsite/src/_11ty/shortcodes/tokensTable/index.js:44]: I think with JSDoc you can even give examples like `token.size`, `token.weight`, hopefully the syntax would allow for `token['text-decoration']` too

File: apps/pie-microsite/test/snapshots/expected-routes.snapshot.json
Comment (jamieomaguire) [inline apps/pie-microsite/test/snapshots/expected-routes.snapshot.json:17]: sorry this is incorrect -it was right before. radius token pages should be inside a /tokens folder

**Review by jamieomaguire — CHANGES_REQUESTED** [review]
we need to fix the routes in the test file

File: apps/pie-microsite/src/_11ty/shortcodes/tokensTable/index.js
Comment (xander-marjoram) [reply apps/pie-microsite/src/_11ty/shortcodes/tokensTable/index.js]: Example from the docs:
```
/**
 * Assign the project to an employee.
 * @param {Object} employee - The employee who is responsible for the project.
 * @param {string} employee.name - The name of the employee.
 * @param {string} employee.department - The employee's department.
 */
Project.prototype.assign = function({ name, department }) {
    // ...
};
```

**Review by FayeCarter — APPROVED** [review]

**Review by jamieomaguire — APPROVED** [review]

**Review by LTurns — APPROVED** [review]

**Pull request merged**
